clab/config.go

```diff
@@ -229,6 +231,21 @@ func (c *CLab) createNodeCfg(nodeName string, nodeDef *types.NodeDefinition, idx
 	if err != nil {
 		return nil, err
 	}
+
+	// download http or https referenced configs
+	if strings.HasPrefix(p, "http://") || strings.HasPrefix(p, "https://") {
+		// download the file to tmp location
+		tmpLoc := filepath.Join(os.TempDir(), "clab-downloads")
+		utils.CreateDirectory(tmpLoc, 0755)
+		filename := fmt.Sprintf("%s-%d.download", nodeCfg.ShortName, time.Now().Unix())
```

Comment:
@steiler I think the filename should be the last element of the http-referenced file concatenated with the date in the format of `YYYY-MM-DD-HH:MM:SS`

Reply:
I'm not sure if I want to agree with that procedure in any case.
To me it made the most sense to have the file downloaded under the name of the node it belongs to. We can argue if the timestamp makes much sense but here is why I chose this approach.

So if you go and download a raw config from e.g. a pastebin (see https://pastebin.com/raw/bFyBKJBU). Then the name makes no sense at all. Anyways we could still take this as the filename. However what if there are GET parameters attached because files come from some other crude inventory tooling. Then the Basefile is all the same and it is the parameters that define the filename.
We could also honor the `Content-Disposition: attachment; filename="filename.jpg"` http-header. But thats not always there and would therefore introduce multiple different naming styles and schemas. My vote is against it.

So reasonable to me is only naming the files after the nodes that they should get applied too. Again, the timestamp we can discuss about.

Does that make sense to you? Any other ideas or concept you have in mind?

Reply:
let's maybe make it `<lab-name>-<node-name>-<file-name>` and ditch timestamp?
timestamp allows you to keep local copies of the remote file, but I don't think this is desired. If the lab topo refers to a remote location that you basically WANT it to follow whatever contents is available remotely, so timestamping is not really needed.

What is needed is a way distinct remote files for nodes from different labs. For that reason lab name and node name are part of the filename.

If the filename is garbage, like ID of sorts, it is not a big deal, you don't really care, since you desired to upload it under that name eventually.

I expect most users using some git service, where file names are part of the fetching URI after all. (most commonly github)